### PR TITLE
Add LR-FHSS data rate coding rates

### LINKED ISSUE #4466: Put coding rate in band's data rate
#### Summary

Declare coding rate in band's data rates

#### Why do we need this?

Unlike with LoRa, with LR-FHSS, the data rate is partly defined by the coding rate.

I think that this is needed for https://github.com/TheThingsNetwork/lorawan-stack/issues/3806

#### What is already there? What do you see now?

Gateway independently sets the PHY's LoRa coding rate on downlink.

With LR-FHSS this isn't possible; it depends on the data rate, but this is not declared in the band.

#### What is missing? What do you want to see?

The coding rate isn't defined in the band.

#### How do you propose to implement this?

I think we should declare the coding rate as part of the LR-FHSS data rate so that it can be get by index.

That is, add it as a field to `message LRFHSSDataRate` API.

We don't have to define the LoRa coding rate in `LoRaDataRate` just yet, let's not do that for now either.

#### How do you propose to test this?

Unit tests to get the full data rate by index and back

#### Can you do this yourself and submit a Pull Request?

Can review

### PR DESCRIPTION
**Pull request by adriansmares**
#### Summary

References #4466
References https://github.com/TheThingsNetwork/lorawan-stack/issues/3806 
References https://github.com/TheThingsNetwork/lorawan-stack/issues/4346

#### Changes

- Add the coding rate to the LR-FHSS data rates (API)
- In GS/PBA use the coding rate from the data rate, when possible

#### Testing

Unit testing.

##### Regressions

None expected.

#### Notes for Reviewers

I've attempted to refactor away the `LoRaCodingRate` from the `band` package and to add the coding rate to the `LoRa` data rate. This is a very invasive change, and must occur in a minor, as a `3.x.y` GS may not send a full LoRa data rate to a `3.x.(y + 1)` PBA. As such, I didn't include it in this PR.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/pfconfig/lbslns/lbslns.go [hunk 1]
```diff
@@ -308,7 +308,6 @@ func getDataRatesFromBandID(id string) (DataRates, error) {
 
 	for i, dr := range phy.DataRates {
 		if loraDR := dr.Rate.GetLora(); loraDR != nil {
-			loraDR.GetSpreadingFactor()
```
Comment (KrishnaIyer) [inline pkg/pfconfig/lbslns/lbslns.go:311]: 🤦

File: pkg/toa/toa.go [hunk 2]
```diff
@@ -144,3 +142,24 @@ func computeFSK(payloadSize int, frequency uint64, bitRate uint32, crc bool) (ti
 		return 0, errFrequency.WithAttributes("frequency", frequency)
 	}
 }
+
+func computeLRFHSS(phyPayloadLength int, codingRate string, crc bool) (time.Duration, error) {
```
Comment (KrishnaIyer) [inline pkg/toa/toa.go:146]: Could you link the spec here if there's a public link?

**Review by KrishnaIyer — APPROVED** [review]
LGTM.

This PR should target the `v3.15` branch right?

**Review by htdvisser — APPROVED** [review]
Code-owned changes good to me 👍

File: pkg/toa/toa.go (hunk 2 above)
Comment (johanstokking) [reply pkg/toa/toa.go:146]: We can do that, but this is defined in LoRaWAN Regional Parameters RP2 1.0.2, not in a specific datasheet like the others. We don't refer to LoRaWAN specs all the time in our codebase, so I guess that's the reason not to mention it.

**Review by johanstokking — APPROVED** [review]
This is more elegant than I could have hoped for

**Comment by johanstokking** [comment]
> I've attempted to refactor away the `LoRaCodingRate` from the `band` package and to add the coding rate to the `LoRa` data rate. This is a very invasive change, and must occur in a minor, as a `3.x.y` GS may not send a full LoRa data rate to a `3.x.(y + 1)` PBA. As such, I didn't include it in this PR.

I think this is fine for now indeed.

**Pull request merged**